### PR TITLE
AnyOf matcher and its tests

### PR DESCRIPTION
**Pull request by mShono**

### CODE REVIEW

File: tests/test_basic.py [hunk 1]
```diff
@@ -22,6 +25,19 @@ def test_dict__plain_match():
     assert {"a": 1, "b": 2} == DictContains({})
 
 
+def test_anyof():
```
Comment (wsnk) [inline tests/test_basic.py:28]: Well, it should be vice versa =)

Actual value should be equal to one of "AnyOf". Not "AnyOf" is equal to one of the actual.

I.e. it should do:
```(python)
assert 1 == AnyOf(1, 2, 3)
```

File: tests/test_basic.py [hunk 2]
```diff
@@ -22,6 +25,19 @@ def test_dict__plain_match():
...
+    assert [8, "a", 3.5] == AnyOf(8)
+    assert [8, "a", 3.5] == AnyOf("a")
+    assert [8, "a", 3.5] != AnyOf(4)
+
+
+def test_similarlist():
+    assert [8, "a", 3.5] == SimilarList(["a", 8, 3.5])
+    assert [8, "a", 3.5] == SimilarList([3.5, 8, "a"])
+    assert [8, "a", 3.5] != SimilarList([3.5, 9, "a"])
+    assert [8, "a", 3.5] != SimilarList([3.5, 8, "a", 10])
```
Comment (wsnk) [inline tests/test_basic.py:38]: Add also test(s) with empty lists, as it's usually a special corner case

File: majava/basic.py [hunk 3]
```diff
@@ -47,3 +51,32 @@ def __eq__(self, other):
...
+        self.value = value
+
+    def __eq__(self, other):
+        return self.value in other
+
+    def __repr__(self):
+        return repr(self.value)
+    
+
+class SimilarList:
```
Comment (wsnk) [inline majava/basic.py:69]: I'd rather name it as "unorderd list" for this particular case.

**Review by wsnk — CHANGES_REQUESTED** [review]

File: majava/basic.py [hunk 4]
```diff
@@ -47,3 +51,32 @@ def __eq__(self, other):
...
         return ', '.join(it.__name__ for it in self.types)
+
+
+class AnyOf:
+    """Actual value must match at least one of the expected values."""
+
+    def __init__(self, value):
+        self.value = value
+
+    def __eq__(self, other):
```
Comment (wsnk) [inline majava/basic.py:62]: Instead of implementing its own `__eq__`, we need to inherite from `Matcher` and implement `_match(self, other)`. Check, how `DictContains` works.
